src/codegen/traverse.lisp

```diff
@@ -250,9 +252,47 @@ other nodes, then it would be inappropriate to also define an action
               (fire-action ':traverse (class-name (class-of current-node)) actions args)
               (fire-action ':after    (class-name (class-of current-node)) actions args)
               (fire-action ':after    'node                                actions args))))
-      (let ((*traverse* #'current-traverse))
+      (let ((*traverse* (lambda (node &rest args)
+                          (funcall *traverse-hook*
+                                   (lambda ()
+                                     (apply #'current-traverse node args))
+                                   node))))
         (apply *traverse* initial-node initial-args)))))
 
+(defun identity-hook (thunk node)
+  "Default traversal hook function, which doesn't do any extra operation."
+  (declare (ignore node)) (funcall thunk))
+
+(defvar *traverse-hook* #'identity-hook)
+
+(defun traverse-with-hook (hook node action-list &rest args)
+  "Like `traverse`, but uses `hook` as the traversal hook function.
+The hook function is called whenever `traverse` recurse into the sub
+nodes.  It takes two arguments: `thunk` to recurse traversal, and `node`
+that is about to be recursed.
```

Comment:
Add a paragraph stating that `identity-traverse-hook` is an identity function, or something like that.

Reply:
`identity-traverse=hook` isn't exported, and it is only called when a bare `traverse` is used.  I can explain that in the documentatino of  `identity-traverse-hook`.

Reply:
You might consider exporting it as a nice "apropos"-able example.

Reply:
That's reasonable.  I'll do so.